projects/storefrontlib/cms-structure/services/guards-composer.ts

```diff
@@ -5,24 +5,31 @@
  */
 
 import { Injectable } from '@angular/core';
-import { CanActivateFn, RouterStateSnapshot, UrlTree } from '@angular/router';
+import {
+  CanActivateFn,
+  GuardResult,
+  RouterStateSnapshot,
+} from '@angular/router';
 import { CmsActivatedRouteSnapshot, wrapIntoObservable } from '@spartacus/core';
 import { Observable, concat, endWith, first, of, skipWhile } from 'rxjs';
 
 /**
  * Replacement for the Angular's deprecated type `CanActivate`.
  */
+//TODO: discuss moving this to `@spartacus/core`
 export type CanActivate = { canActivate: CanActivateFn };
 
 /**
  * Observable that emits a boolean or an UrlTree.
  */
-export type CanActivateObservable = Observable<boolean | UrlTree>;
+//TODO: discuss moving this to `@spartacus/core`
+export type CanActivateObservable = Observable<GuardResult>;
 
 /**
  * Utility service for running multiple guards and composing their results
  * into a single result observable.
  */
+//TODO: discuss moving this to `@spartacus/core`
```

Comment:
What arguments do you have for/against moving it to spartacus/core?

Reply:
Apologies, after rethinking it, the proposal could applies only to:
```ts
export type CanActivateObservable = Observable<GuardResult>;
```
because I've seen the usage of type `Observable<GuardResult>` in many places across all libs, where `CanActivateObservable` could be used. But my second thoughts are KISS instead of DRY and don't create unnecessary dependency.